tests/store/artifact/test_uc_volumes_artifact_repo.py

```diff
@@ -0,0 +1,222 @@
+import posixpath
+from unittest import mock
+
+import pytest
+
+from mlflow.entities.file_info import FileInfo
+from mlflow.exceptions import MlflowException
+from mlflow.store.artifact.artifact_repository_registry import get_artifact_repository
+from mlflow.store.artifact.uc_volume_artifact_repo import UCVolumesArtifactRepository
+
+HOST = "http://localhost:5000"
+
+
+def join_non_empty(*args):
+    """
+    Join path components, ignoring empty components.
+    """
+    non_empty_args = [a for a in args if a]
+    return posixpath.join(*non_empty_args)
+
+
+@pytest.fixture(autouse=True)
+def set_creds(monkeypatch):
+    monkeypatch.setenv("DATABRICKS_HOST", "http://localhost:5000")
+    monkeypatch.setenv("DATABRICKS_TOKEN", "abc")
+
+
+@pytest.fixture
+def artifact_repo(monkeypatch):
+    return get_artifact_repository("dbfs:/Volumes/catalog/schema/volume/run_id/artifacts")
+
+
+@pytest.mark.parametrize(
+    "artifact_uri",
+    [
+        "dbfs:/Volumes/catalog/schema/volume/path",
+        "dbfs:/volumes/catalog/schema/volume/path",
+        "dbfs:/Volume/catalog/schema/volume/path",
```

Comment:
+1 to validating the typo-aware match logic in the UC implementation

Reply:
This is not a typo. `Volume`, `volume`, and `volumes` are all alias:

https://docs.databricks.com/en/volumes/index.html#reserved-paths-for-volumes